src/redux/missions/missions.js

```diff
@@ -0,0 +1,53 @@
+const url = 'https://api.spacexdata.com/v3/missions';
+
+// ACTIONS
+const SET_MISSION = 'missionsStore/missions/SET_MISSION';
+const RESERVE_MISSION = 'missionsStore/missions/RESERVE_MISSION';
+
+const initialState = {
+  missions: [],
+};
+
+// ACTIONS CREATORS
+
+export const missionArray = (response) => {
+  response.map((mission) => (
+    initialState.missions.push({
+      id: mission.mission_id,
+      name: mission.mission_name,
+      description: mission.description,
+    })));
+  return initialState;
+};
+
+export const missionReserve = (payload) => ({
+  type: RESERVE_MISSION,
+  payload,
+});
+
+export const setMission = () => async (dispatch) => {
+  fetch(url)
+    .then((rawResponse) => rawResponse.json())
+    .then((response) => dispatch({
+      type: SET_MISSION,
+      payload: missionArray(response),
+    }));
+};
```

Comment:
**Optional Change**
- [ ] Fair effort on using the `async await`  promise syntactic sugar. However, you're mixing two syntax here, and the `async` keyword in this context is redundant. May we please remove the `.then` and stick to the `async await` format.